### DISCUSSION TITLE
Frosted-Glass Theme

### DISCUSSION THREAD

**Discussion by T-Crypt** [prasanthrangan/hyprdots#117] [Ideas]
@prasanthrangan 

I have started working on the Frosted-Glass theme on my fork, feel free to grab anything you need. It's a work in progress but stable since you have the main branch.

https://github.com/T-Crypt/hyprdots

**Comment by prasanthrangan** [comment]
Hi @T-Crypt , Thank you so much!
I'm not working on any theme currently, so once you are done with Frosted-Glass theme on your fork, please feel free to create a pull request for it.
Then I will approve and merge it to this repo.

**Comment by T-Crypt** [comment]
@prasanthrangan 

I am almost done with the theme.

Where do I find the dolphin font color to change from black to white?

And how do I get the images cached into the wallpaper selector or theme selector without first selecting it? Once its selected, it is cached, but upon install it is blank! THANK YOU :)

![230828_10h25m26s_screenshot](https://github.com/prasanthrangan/hyprdots/assets/132922589/f69b8be9-d759-45ae-b32b-951c2e467d6f)
![230828_10h26m22s_screenshot](https://github.com/prasanthrangan/hyprdots/assets/132922589/d7a85fcf-1f9e-4fa0-8630-5be2d322124b)
![230828_10h24m40s_screenshot](https://github.com/prasanthrangan/hyprdots/assets/132922589/132b1bb0-13f4-4123-b7d0-39d4ed0b5209)


Also, hope the color-scheme is okay for you, I saw you added Nordic-Blue but most of the frosted glass themes came across close to the catppuccin-latte theme. Figured Nordic-Blue can be a dark-blue theme

**Comment by prasanthrangan** [comment]
> I am almost done with the theme.

Woww... looks dope!

> Where do I find the dolphin font color to change from black to white?

Its in both `~/.config/qt5ct/colors/Frosted-Glass.conf` and `~/.config/Kvantum/Frosted-Glass/Frosted-Glass.kvconfig`

> And how do I get the images cached into the wallpaper selector or theme selector without first selecting it? Once its selected, it is cached, but upon install it is blank! THANK YOU :)

Yup, that's expected behavior, it will generate the wallpaper cache when you use the wallpaper for the first time, however @Kemipso has already contributed #124 a script to fix this during install... so as part of restore script, when you apply new updates it will generate the wallpaper cache!

> Also, hope the color-scheme is okay for you, I saw you added Nordic-Blue but most of the frosted glass themes came across close to the catppuccin-latte theme. Figured Nordic-Blue can be a dark-blue theme

The light blue color scheme looks amazing... only suggestion would be to maintain the same color consistency for all apps/configs!
You could also try messing around with soft shadows and adding more transparency and rounding, I'm guessing it would make your theme look more like frosted glass.

**Comment by T-Crypt** [comment]
@prasanthrangan 

I've placed the Frosted-Glass (GTK Theme) as a tar.gz file in "arc" source. 

Where is your function to extract the tar.gz files that contain the GTK themes and place in ./themes? Im trying to modify it to include the Gtk_Frosted-Glass.tar.gz directory and place in ./themes in the ./install.sh script 

Also, wanted to know where you called the VScode theme install / placement as I found this theme works best for Frosted-Glass. 

![Screenshot 2023-08-28 150048](https://github.com/prasanthrangan/hyprdots/assets/132922589/a8684d95-08d7-4e5d-9f71-07d94ee820f3)

**Reply by T-Crypt** [reply]
latest update

![theme_frosted_1](https://github.com/prasanthrangan/hyprdots/assets/132922589/520a5f2d-ab4d-49dc-b818-e5247ba6fe47)
![theme_frosted_2](https://github.com/prasanthrangan/hyprdots/assets/132922589/1e6c72a1-27b5-434f-9a9c-80ff67a6f569)

**Comment by prasanthrangan** [comment]
Nice progress, looks on point!!

> Where is your function to extract the tar.gz files that contain the GTK themes and place in ./themes? Im trying to modify it to include the Gtk_Frosted-Glass.tar.gz directory and place in ./themes in the ./install.sh script

its in `restore_fnt.lst`, just place the `Gtk_Frosted-Glass.tar.gz` in `Source/arcs/` and add these lines 

```
Gtk_Frosted-Glass|/usr/share/themes
Gtk_Frosted-Glass|$HOME/.themes
```

> Also, wanted to know where you called the VScode theme install / placement as I found this theme works best for Frosted-Glass.

for Vs code theme you'll have to repackage the `Source/arcs/Code_Extensions.tar.gz` with `blue light theme` extension folder

**Reply by T-Crypt** [reply]
What modification is needed for the extensions.json? 

I'm about to repackage the extensions and update the arc sources and Ill push a pull request for review.

**Comment by DiegoSof2** [comment]
> @prasanthrangan
> 
> I have started working on the Frosted-Glass theme on my fork, feel free to grab anything you need. It's a work in progress but stable since you have the main branch.
> 
> https://github.com/T-Crypt/hyprdots

I don't know if it's a bug or the icons haven't been added yet, but here the icons don't appear in dolphin , and his icon is not displayed in the waybar
![image](https://github.com/prasanthrangan/hyprdots/assets/105827698/cd754de8-d933-47b4-a262-cdb6239afe32)





But I loved the theme it was very good with a great choice of colors

**Reply by prasanthrangan** [reply]
It's just the missing icon pack... I'll add it

**Comment by prasanthrangan** [comment]
Hi @T-Crypt ,
Thanks for your time and effort in creating and contributing your theme!!

I did few changes for Frosted-Glass,
1. adjusted blur
2. fixed vscode extension
3. add blue icons
4. redid the kvantum config as it was dark and other apps were white.
5. replaced few wallpapers as it looked a bit grainy

Hope this look good,
![230831_00h04m54s_screenshot](https://github.com/prasanthrangan/hyprdots/assets/106020512/aef15381-8538-4bd1-9bb9-23b7d3817cab)

**Reply by T-Crypt** [reply]
@prasanthrangan 

It's perfect! Thank you for your help!!

**Comment by hoodust** [comment]
@T-Crypt Just wanna say I love this theme and thank you. I use it as my daytime theme and switch to Tokyo Night after dark.

qq tho: I use Brave browser and this theme makes it transparent also, and yet I can't manage to do that in any other theme using windowrules.conf... do you know how can I make Brave transparent/opaque at will, preferably with a keybinding? I'm confused because this theme still respects the opacity values present in windowrules.conf but it's doing something differently.

**Reply by hoodust** [reply]
Ah, found it. It's just the global opacity settings in the theme .conf file that the other themes don't use
```
    active_opacity = 0.75
    inactive_opacity = 0.75
```
I suppose I could keybind a script that toggles replacing all the confs with versions that either have or don't have these lines.

**Reply by kRHYME7** [reply]
Hello @hoodust . Additional info theme.conf is a symlink of the current $Theme-name.conf  So it will change. 

I am not sure yet but maybe you can override that config by adding to the userprefs and from that maybe Toggle it using a script?

**Reply by hoodust** [reply]
Sorry, I meant the original Frosted-Glass.conf file. I can change it there and it automagically applies it live (thanks to the symlink)

**Comment by Kemipso** [comment]
Hey @hoodust ,

You could use the userprefs.conf file, and call hyprctl keywords for example (without script), which allows you to adjust configs on the fly.
See http://wiki.hyprland.org/Configuring/Using-hyprctl/#keyword

Here's a mediocre example (keybindings are a bit clunky), but it conveys what you could do:
```
# Example for https://github.com/prasanthrangan/hyprdots/discussions/117
bind = $mainMod ALT, T, submap, transparency

# In this submap, only 3 shortcuts are left for your Hyprland: up (set opacity to 1 + exit submap), down (set opacity to 0.75 + exit submap), and escape (no change, exit submap).
submap=transparency
bind=,up, exec, hyprctl keyword decoration:active_opacity 1.00
bind=,up, exec, hyprctl keyword decoration:inactive_opacity 1.00
bind=,up, submap, reset
bind=,down, exec, hyprctl keyword decoration:active_opacity 0.75
bind=,down, exec, hyprctl keyword decoration:inactive_opacity 0.75
bind=,down, submap, reset
bind=,escape, submap, reset

submap=reset
```
... Oh, and obviously, you could also just have two shortcuts to adjust the opacity without submap, or you could instead call a bash script if you don't like hyperctl, etc etc etc :-)

**Reply by Kemipso** [reply]
And I just now realize that using this snippet from the official wiki might be even neater, depending on what you'd like to toggle:
http://wiki.hyprland.org/Configuring/Uncommon-tips--tricks/#toggle-animationsbluretc-hotkey

**Reply by hoodust** [reply]
Ah, very cool. Your mapping idea works perfectly, but you're right, that snippet brings up a great way to toggle so you don't need two mappings. 

```
#!/usr/bin/env sh
HYPRBLURALL=$(hyprctl getoption decoration:active_opacity | awk 'NR==3{print $2}')
if [ "$HYPRBLURALL" = "1.00000" ] ; then
    hyprctl --batch "\
        keyword decoration:active_opacity 0.80;\
        keyword decoration:inactive_opacity 0.80"
    exit
else
    hyprctl --batch "\
        keyword decoration:active_opacity 1.00;\
        keyword decoration:inactive_opacity 1.00"
    exit
fi
hyprctl reload
```

`bind = $mainMod SHIFT, B, exec, ~/hyprblurall.sh`

As a bonus, TIL about Shimeji from the gif above that snippet XD

Thanks so much!!
